designs/static-capacity.md

```diff
@@ -0,0 +1,202 @@
+As a result, even though all NodeClaims come from the same static NodePool, they may still result in different instance types (shapes/flavors), depending on availability, since that decision happens during cloud-provider Create() call.
+
+To ensure high availability and fault tolerance, we automatically spread static nodes across AZs when the zone requirement includes multiple values.
+```yaml
+- key: topology.kubernetes.io/zone
+  operator: In
+  values: ["us-west-2a", "us-west-2b", "us-west-2c"]
+```
+
+This ensures the nodes are evenly distributed unless the user explicitly narrows the zone selection to a single AZ.
+
+## Example 
+
+```yaml
+apiVersion: karpenter.sh/v1beta1
+kind: NodePool
+metadata:
+  name: static-prod-nodepool
+spec:
+  replicas: 12 # Maintains 12 nodes regardless of pod demand
+  template:
+    metadata:
+      labels:
+        nodepool-type: static
+    spec:
+      nodeClassRef:
+        group: karpenter.k8s.aws
+        kind: EC2NodeClass
+        name: myEc2NodeClass
+      expireAfter: 720h
+      taints:
+        - key: example.com/special-taint
+          effect: NoSchedule
+      requirements:
+        - key: topology.kubernetes.io/zone
+          operator: In
+          values: ["us-west-2a", "us-west-2b", "us-west-2c"]
+        - key: karpenter.k8s.aws/instance-type
+          operator: In
+          values: ["m5.2xlarge"]
+  disruption:
+    budgets:
+      - nodes: 10%
+      # On Weekdays during business hours, don't do any deprovisioning.
+      - schedule: "0 9 * * mon-fri"
+        duration: 8h
+        nodes: "0"
+  limits:
+    # Constrain the size of static Nodepool 
+    nodes: 20 # Karpenter will not scale up nodes or create replacements if limits are breaching
+```
+
+We will support scaling static node pools using a command such as:
+
+```sh
+kubectl scale nodepool <name> --replicas=10
+```
+
+## Implementation Details
+
+### Controller Architecture
+
+We will be creating Static Capacity by adding 2 new controllers under a feature flag. This will be added to v1alpha1 release to get feedback from users.
+
+#### Scale-Up Controller
+- **Purpose**: Scale up nodeclaims to meet desired replicas
+- **Behavior**: Monitors the difference between desired replicas and current healthy NodeClaims, provisioning new nodes as needed respecting limits.nodes value
+- **Instance Selection**: Will find the cheapest instance possible during scale up operations based on the requirements
+- **Consolidation**: Static NodePools will not be consolidated to maintain predictable capacity
+
+#### Deprovisioning Controller  
+- **Purpose**: Scale down replicas when desired count is reduced
+- **Node Selection**: Will prioritize empty nodes for termination. In future iterations, we will add intelligent algorithms for selecting which nodes to scale down
```

Comment:
I take it if there aren't non-empty nodes it will be random? Let's be explicit about the proposed selection logic.